assets/src/components/form/validations/index.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * External imports
+ */
+import { __, _n, sprintf } from '@eventespresso/i18n';
+
+/**
+ * @function
+ * @param {mixed} value
+ * @return {string|null} error string on fail
+ */
+export const isFloat = value => {
+	// first convert to string
+	value = value + '';
+	// convert decimal to known format
+	value = value.replace( ',', '.' );
+	// strip all trailing zeros
+	value = value.replace( /0+$/, '' );
+	const lastChar = value.substr( value.length - 1 );
+	const float = (
+		// parsed value is same as original coerced to a number
+		parseFloat( value, 10 ) === +value &&
+		(
+			// no remainder but last character is a decimal
+			// ie: original value like 1.0
+			( value % 1 === 0 && lastChar === '.' ) ||
+			// has a remainder and last character is NOT a decimal
+			// ie: original value like 1.# where # is not 0
+			( value % 1 !== 0 && lastChar !== '.' )
+		)
+	);
+	return ! float ? __( 'Value must be a float', 'event_espresso' ) : null;
+};
+
+/**
+ * @function
+ * @param {mixed} value
+ * @return {string|null} error string on fail
+ */
+export const isInteger = value => {
+	return ! ( parseInt( value, 10 ) === +value || value % 1 === 0 ) ?
+		__( 'Value must be an integer', 'event_espresso' ) :
+		null;
+};
+
+/**
+ * @function
+ * @param {mixed} value
+ * @return {string|undefined} error string on fail
+ */
+export const isNumeric = value => {
+	return isNaN( value ) ?
+		__( 'Value must be a number', 'event_espresso' ) :
+		null;
+};
+
+/**
+ * @function
+ * @param {mixed} field
+ * @param {string} fieldName
+ * @return {string|undefined} error string on fail
+ */
+export const matches = ( field, fieldName ) => value => {
+	return ! value || field !== value ?
+		sprintf(
+			__( 'Value should match %s', 'event_espresso' ),
```

Comment:
Would `__( 'Field value should match %s', 'event_espresso' )` be a bit better?

Reply:
no my concern is that, although devs and any technically minded people would understand what is meant by "field" there are many others who simply would not.

But if presented with an input labelled "email address" and another labelled "confirm email", and an error message under the latter saying: "Value should match email address" I don't think there will be any confusion as to what is required.

Reply:
Ahh, I was ignorant of the context this would be displayed in.  You're right 👍